app/module/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -318,6 +319,28 @@ static int kscan_direct_init(const struct device *dev) {
     return 0;
 }
 
+#if IS_ENABLED(CONFIG_PM_DEVICE)
+
+static int kscan_direct_pm_action(const struct device *dev, enum pm_device_action action) {
+    int ret = 0;
+
+    switch (action) {
+    case PM_DEVICE_ACTION_SUSPEND:
+        kscan_direct_disable(dev);
+        break;
+    case PM_DEVICE_ACTION_RESUME:
+        kscan_direct_enable(dev);
+        break;
+    default:
+        ret = -ENOTSUP;
+        break;
+    }
+
+    return ret;
```

Comment:
Minor: this is ignoring the return code for enable/disable. Could fix that and simplify a bit by just using `return` in the switch?

```suggestion
    switch (action) {
    case PM_DEVICE_ACTION_SUSPEND:
        return kscan_direct_disable(dev);

    case PM_DEVICE_ACTION_RESUME:
        return kscan_direct_enable(dev);

    default:
        return -ENOTSUP;
    }
```